torch/csrc/autograd/engine.cpp

```diff
@@ -75,6 +76,25 @@ inline bool should_run_in_cpu_ready_queue(c10::DeviceType device) {
     return false;
   }
 }
+
+static std::atomic<Engine::compiled_autograd_fn> the_compiled_autograd =
+    nullptr;
+static std::atomic<int32_t> num_threads_in_backwards;
+struct CompiledAutogradThreadingDebugCheck {
+  CompiledAutogradThreadingDebugCheck() : active(true) {
+    num_threads_in_backwards++;
+  }
+  ~CompiledAutogradThreadingDebugCheck() {
+    release();
+  }
+  void release() {
+    if (std::exchange(active, false)) {
+      num_threads_in_backwards--;
+    }
+  }
+  bool active;
```

Comment:
Since this struct has invariants that must be upheld, it would be better for the field to be private. Just makes subsequent invariant enforcement easer...